src/main/java/de/kekru/struktogrammeditor/other/SupportedLanguages.java

```diff
@@ -0,0 +1,78 @@
+package de.kekru.struktogrammeditor.other;
+
+/**
+*
+* Struktogrammeditor
+*
+* @version Unbekannt
+* @author Kevin Krummenauer
+* Initialer Code
+* 
+* @version 1.7.4
+* @author Rafael Sundorf
+* Umstellung auf Enum und anderen Anpassung
+*/
+public enum SupportedLanguages {
+	
+	Java ("Java", 0),
+	Delphi ("Delphi", 1),
+	PHP ("PHP", 2);
+	
+	
+	private String name;
+	private int id;
+	private SupportedLanguages (String name, int id) {
+		this.name = name;
+		this.id = id;
+	}
+	
+	public String getName () {
+		return this.name;
+	}
+	
+	
+	public int getId () {
+		return this.id;
+	}
+	
+	public static SupportedLanguages getById (int id) {
+		for (SupportedLanguages l : values()) {
```

Comment:
With java 8 Streams it would be

```suggestion
    return Stream.of(values())
        .filter(entry -> entry.getId() == id)
        .findFirst()
        .orElseThrow(() -> new RuntimeException("No SupportedLanguages found with id " + id))
```

And better throw exception instead of return null